compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java

```diff
@@ -583,7 +583,7 @@ private void rewriteObservableFunctionInvocations(BIRFunction func, BIRPackage p
                         objectTypeOperand = generateGlobalConstantOperand(pkg, symbolTable.nilType, null);
                         action = callIns.name.value;
                     }
-                    currentBB.terminator = new GOTO(desugaredInsPosition, observeStartBB);
+                    currentBB.terminator = new GOTO(desugaredInsPosition, observeStartBB, null);
```

Comment:
Shall we have multiple constructors for these classes to avoid passing null?

Reply:
Shall we fix in all places?

Reply:
If we want to do so we need to consider other already existing parameters also. For example we are passing null for the position here https://github.com/ballerina-platform/ballerina-lang/blob/d4eb62a39664bbee06b2a624751bd453d3a5c9c2/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java#L700 So we might end up with several constructors with different combinations. IMO, it is good to keep them as it instead of having multiple constructors. WDYT?

Reply:
Normally we try to reduce technical debt in the new code that we write irrespective of the code already there. If we can improve the old code that is fine, but in the new code it is good to follow the best practices.